### ISSUE TITLE
Auto-Import

### ISSUE THREAD

**Issue by ExTheSea** [stackblitz/core#60]
First of all, this is an amazing project for prototyping Web applications or developing applications on devices not capable of handling full-fletched IDE's (e.g. Raspberrys, Tablets, etc.). I'm also heavily looking forward to #9 as it's otherwise quite tedious to create new components/services/etc.

Another suggestion that I couldn't find (or i've just missed) is a better auto-import:
- Auto-complete can't find Modules from dependencies you added (like the angular material modules)
- When creating a new component right now and adding it to the declaration array in the AppModule the editor can't find the new component
- Even the angular classes like e.g. Component are not being found
- Not thoroughly tested but it seems that the React classes like Component are not available through Intellisense

For all of those you have to manually add the ``import { ... } from './....ts';`` manually and without any file system auto-complete this can get quite tedious, especially when importing something from an external dependency.

I'm quite used to an auto-import behaviour from IntelliJ but even VSCode can support something like this with plugins like [Auto-Import](https://marketplace.visualstudio.com/items?itemName=steoates.autoimport).

So as you were mentioning natively implementing VSCode plugin behaviour and what functionally is needed in #3 I feel this would be an important addition if possible.

**Comment by EricSimons** [comment]
@ExTheSea I totally agree & really love this idea. It's super annoying to have to guess & check whether you're importing the right thing from the right place all the time 😒

@claytonschneider wanna take this on?

**Comment by kordeviant** [comment]
this is a way of exponential growth.

**Comment by Splaktar** [comment]
This is certainly a significant need. It makes learning with StackBlitz quite a bit harder to get started with and makes experienced devs less productive.

**Comment by bastienJS** [comment]
come on guys nearly a year later...

**Comment by briancodes** [comment]
There's an online Angular editor called https://ng-run.com/, which looks like it's based on VSCode. It has an auto-import feature which might be something that could be added to StackBlitz - putting the link here for inspiration. StackBlitz is really top notch, and auto-import would definetly speed up development (and minimize time spent fixing import errors😉)

**Comment by andreElrico** [comment]
any updates?

**Comment by vargalas** [comment]
I've just paid for Stackblitz, but I won't be able to use it without auto-import.

**Comment by d3lm** [comment]
@vargalas @andreElrico Yep, we are working on auto-imports!

**Comment by dnkm** [comment]
> 
> 
> @vargalas @andreElrico Yep, we are working on auto-imports!

As soon as you guys have auto-imports and auto-complete for React, you have my business!

**Comment by afreenCodez** [comment]
auto import added to stackblitz ?

**Comment by bfunc** [comment]
auto import is still missing, Stackblitz is much faster than Codesandbox but this feature is really missing

**Comment by leoncvlt** [comment]
Huge +1 for this option, Stackblitz seems to give me the least amount of bugs / side-effects compared to replit and Codesandbox, but without this feature my productivity really takes a hit, especially when working with libraries with a huge amount of exports like three.js

https://github.com/stackblitz/monaco-auto-import - Does this repo have anything to do with it? Hasn't been updated since 2018...

**Comment by kj6995** [comment]
Any luck in 2022?

**Comment by arrbxr** [comment]
5 years on and we still don't have auto imports :(

**Comment by MarsOrBust24** [comment]
FYI: Auto-import is working for me now on typescript files. I tested by going to [vite.new/vue-ts](https://vite.new/vue-ts), opening `src/main.ts` and typing `const x = reactive([])` that automatically suggested reactive from Vue and added it to my imports. It seems to also work on `.mjs` scripts and `type: "module"` node projects.


Unfortunately It doesn't work on Vue/Volar script setup Vue 3 ts SFC components.

**Comment by leoncvlt** [comment]
Still doesn't work with vanilla JS packages though.

For example, if I add `three` as a dependency, I get auto-completion for any of the name exports defined here https://github.com/mrdoob/three.js/blob/dev/src/Three.js when developing this locally with VS code, or using CodeSandbox or Replit - but not with Stackblitz 😔

I'd rather have this fixed than receive emails about viteconf 👀

**Comment by kc0tlh** [comment]
With the addition of VS Code extension support in our new editor, this functionality can be achieved through the [Auto-Import](https://marketplace.visualstudio.com/items?itemName=steoates.autoimport) extensions mentioned by the OP @ExTheSea!

<img width="1438" alt="image" src="https://github.com/stackblitz/core/assets/1532264/3ed0e186-16dd-431e-bbc4-0ad9a2411798">

**Comment by MoinShaikh611** [comment]
> 5 years on and we still don't have auto imports :(


6 years on and we still don't have auto imports :(

**Comment by akashbharati** [comment]
why such a long wait for auto-import ?
